media/js/base/stub-attribution.js

```diff
@@ -575,7 +576,107 @@ if (typeof window.Mozilla === 'undefined') {
                     }
                 }
             });
+
+            // Wait for GTM to load the GTAG GET API so that we can pass along client and session ID
+            StubAttribution.waitForGTagAPI(function () {
+                gtagData = StubAttribution.getGtagData();
+
+                if (gtagData && StubAttribution.withinAttributionRate()) {
+                    // GA4 - send an event indicating we did or did not get session ID from GTM
+                    if (gtagData.client_id && gtagData.session_id) {
+                        window.dataLayer.push({
+                            event: 'stub_session_test',
+                            label: 'session found'
+                        });
+                    } else {
+                        window.dataLayer.push({
+                            event: 'stub_session_test',
+                            label: 'session not-found'
+                        });
+                    }
+                }
+            });
+        }
+    };
+
+    /**
+     * Attempts to retrieve the GA4 client and session ID from the dataLayer
+     * The GTAG GET API tag will write it to the dataLayer once GTM has loaded it
+     * https://www.simoahava.com/gtmtips/write-client-id-other-gtag-fields-datalayer/
+     */
+    StubAttribution.getGtagData = function (dataLayer) {
+        dataLayer =
+            typeof dataLayer !== 'undefined' ? dataLayer : window.dataLayer;
+
+        var result = {
+            client_id: null,
+            session_id: null
+        };
+
+        var _findObject = function (name, obj) {
+            for (var key in obj) {
+                if (
+                    typeof obj[key] === 'object' &&
+                    Object.prototype.hasOwnProperty.call(obj, key)
+                ) {
+                    if (key === name) {
+                        if (
+                            typeof obj[key].client_id === 'string' &&
+                            typeof obj[key].session_id === 'string'
+                        ) {
+                            result = obj[key];
+                        } else {
+                            return result;
+                        }
+                        break;
+                    } else {
+                        _findObject(name, obj[key]);
+                    }
+                }
+            }
+        };
+
+        if (typeof dataLayer === 'object') {
+            dataLayer.forEach(function (layer) {
+                _findObject('gtagApiResult', layer);
+            });
         }
+
+        return result;
```

Comment:
Suggestion: I wonder if it's worth wrapping this in a `try/catch`, just to be safe seeing as we're parsing third party code?